marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/mapping-functions/core.sjs

```diff
@@ -137,7 +138,11 @@ function parseDate(value, pattern) {
     }
   }
   else {
-    fn.error(null, "The given date pattern (" + pattern + ") is not supported.");
+    try {
+      response = xs.date(xdmp.parseYymmdd(originalPattern, value));
+    } catch(e) {
+      fn.error(null, "The given date pattern (" + originalPattern + ") is not supported.");
```

Comment:
The story mentions that every error should be of the form - "Also, every Error result should be a message of "The pattern '$pattern' cannot be applied to the value '$value'"." - is that possible to do - which includes not showing the XSLT path where it failed, which a commenter on the story described as "user hostile"?